truffle/src/main/java/org/jruby/truffle/runtime/subsystems/ThreadManager.java

```diff
@@ -73,13 +69,7 @@ public void enterGlobalLock(RubyThread thread) {
      */
     @TruffleBoundary
     public RubyThread leaveGlobalLock() {
-        if (!globalLock.isHeldByCurrentThread()) {
-            throw new RuntimeException("You don't own this lock!");
-        }
-
-        final RubyThread result = currentThread;
-        globalLock.unlock();
-        return result;
+        return currentThread.get();
```

Comment:
Same question as with `enterGlobalLock`: should this method be renamed if there's no longer a global lock?